elytron/src/main/java/org/wildfly/extension/elytron/ProviderDefinitions.java

```diff
@@ -199,12 +202,20 @@ protected boolean dependOnProviderRegistration() {
                                     } else if (configSupplier != null) {
                                         // Try and find a constructor that takes an input stream and use it - or use default constructor and enable deferred initialisation.
                                         try {
-                                            Constructor<? extends Provider> constructor = doPrivileged((PrivilegedExceptionAction<Constructor<? extends Provider>>) () ->  providerClazz.getConstructor(InputStream.class));
-                                            loadedProviders.add(constructor.newInstance(new Object[] { configSupplier.get() }));
-                                        } catch (NoSuchMethodException ignored) {
-                                            Provider provider = providerClazz.newInstance();
-                                            loadedProviders.add(provider);
-                                            deferred.add(provider::load);
+                                            Constructor<? extends Provider> constructor = doPrivileged((PrivilegedExceptionAction<Constructor<? extends Provider>>) () -> providerClazz.getConstructor(InputStream.class));
+                                            loadedProviders.add(constructor.newInstance(new Object[]{configSupplier.get()}));
+                                        } catch (NoSuchMethodException constructorDoesNotExist) {
+                                            try {
+                                                Method configureMethod = doPrivileged((PrivilegedExceptionAction<Method>) () -> providerClazz.getMethod("configure", String.class));
+                                                Provider provider = providerClazz.newInstance();
+                                                File tempFile = inputStreamToFile(configSupplier);
+                                                loadedProviders.add((Provider) configureMethod.invoke(provider, tempFile.getAbsolutePath()));
+                                                tempFile.delete();
```

Comment:
It might be good to move the deletion to a `finally` block.

Reply:
@fjuma yes thank you. Fixed